### ISSUE TITLE
Add support for calendar-style recurring datetimes to Interval

### ISSUE THREAD

**Issue by bitwalker** [bitwalker/timex#129]
This would permit expressing intervals like "the third Sunday of every month"

**Comment by matheus-b** [comment]
Dear bitwalker, I am implementing recurring events for a phoenix project using timex. Is this the right place to ask you about your thoughts on future compatibility? I basically follow the iCalendar feature set currently.

**Comment by bitwalker** [comment]
Yes definitely :), feel free to post your thoughts/questions here and we can discuss the implementation

**Comment by shymega** [comment]
Just wondering how progression of this feature has been going. I've been trying to implement a similar thing to find the 2nd Saturday of each month, and I'd rather use Timex to do that, but I'm not exactly sure how to implement it either....

**Comment by matheus-b** [comment]
Ah, I forgot about this there.

I could give it a go, have code around here somewhere.

But I am not so sure anymore timex is the place this should be in.

What are your thoughts by now bitwalker?

**Comment by bgentry** [comment]
I'd love to see this. I started using the [recurring_events](https://github.com/pbogut/recurring_events) library, but soon discovered that it does not handle DST transitions at all because it doesn't have any real awareness of time zones or daylight saving time (https://github.com/pbogut/recurring_events/issues/14).

If the goal is to have an event that repeats "at 3am every day", that should be done in a way that respects DST transitions in the timezone. Or at least there should be an option to do so, because I can understand there may be cases where this is not desirable.
